packages/mco/components/disaster-recovery/applications/failover-and-relocate-modal/modal-body.tsx

```diff
@@ -0,0 +1,335 @@
+            payload: {
+              drPolicyControlStateErrorMessage: t(
+                'Cannot initiate {{actionType}} as DR is not enabled, apply DRPolicy to the application to proceed.'
+              ),
+            },
+          });
+    }
+  }, [
+    drPlacementControls,
+    drPlacementControlsLoaded,
+    placementRulesLoaded,
+    subscriptionsLoaded,
+    drPlacementControlsLoadError,
+    placementRulesLoadError,
+    subscriptionsLoadError,
+    subscriptionMap,
+    action,
+    dispatch,
+    t,
+  ]);
+
+  return (
+    <Flex
+      direction={{ default: 'column' }}
+      spaceItems={{ default: 'spaceItemsSm' }}
+      className="mco-dr-action-body__line mco-dr-action-body__flex"
+    >
+      <Flex>
+        <FlexItem>
+          <strong> {t('Application name:')} </strong>
+        </FlexItem>
+        <FlexItem>
+          <ResourceIcon resourceModel={ApplicationModel} />{' '}
+          {application?.metadata?.name || '-'}
+        </FlexItem>
+      </Flex>
+      <Flex
+        direction={{ default: 'column' }}
+        spaceItems={{ default: 'spaceItemsSm' }}
+      >
+        <FlexItem>
+          <strong> {t('Select policy')} </strong>
+        </FlexItem>
+        <FlexItem>
+          <DRPolicySelector state={state} dispatch={dispatch} />
+        </FlexItem>
+      </Flex>
+      <Flex
+        direction={{ default: 'column' }}
+        spaceItems={{ default: 'spaceItemsSm' }}
+      >
+        <FlexItem>
+          <strong> {t('Target cluster')} </strong>
+        </FlexItem>
+        <FlexItem>
+          <TargetClusterSelector state={state} dispatch={dispatch} />
+        </FlexItem>
+      </Flex>
+      <PeerStatus state={state} dispatch={dispatch} action={action} />
+      <Flex
+        grow={{ default: 'grow' }}
+        direction={{ default: 'column' }}
+        spaceItems={{ default: 'spaceItemsSm' }}
+      >
+        <FlexItem>
+          <strong> {t('Select subscriptions group')} </strong>
+        </FlexItem>
+        <FlexItem>
+          <SubscriptionGroupSelector state={state} dispatch={dispatch} />
+        </FlexItem>
+      </Flex>
+      {(!!findErrorMessage(state.errorMessage) &&
+        getMessageStatus(
+          findErrorMessage(state.errorMessage),
+          AlertVariant.danger
+        )) ||
+        (state.modalFooterStatus === ModalFooterStatus.FINISHED &&
+          getMessageStatus(
+            t('{{actionType}} initiated', {
+              actionType: state.actionType,
+            }),
+            AlertVariant.success
+          ))}
+    </Flex>
+  );
+};
+
+type ModalBodyProps = {
+  application: ApplicationKind;
+  action: ACTION_TYPE;
+  state: FailoverAndRelocateState;
+  dispatch: React.Dispatch<FailoverAndRelocateAction>;
+};
+
+type DRActionWatchResourceType = {
+  placementRules: ACMPlacementRuleKind[];
+  subscriptions: ACMSubscriptionKind[];
+  drPlacementControls: DRPlacementControlKind[];
+};
+
+type SubcsriptionMap = {
+  [placementRuleName in string]: string[];
```

Comment:
```suggestion
  [placementRuleName: string]: string[];
```
nit: use in when you are working with enums or interfaces.